code/framework/encryption/src/main/java/io/cattle/platform/framework/encryption/request/handler/DeTransformationHandler.java

```diff
@@ -0,0 +1,70 @@
+package io.cattle.platform.framework.encryption.request.handler;
+
+import io.cattle.platform.framework.encryption.handler.TransformerHandler;
+import io.github.ibuildthecloud.gdapi.factory.SchemaFactory;
+import io.github.ibuildthecloud.gdapi.model.Field;
+import io.github.ibuildthecloud.gdapi.model.Resource;
+import io.github.ibuildthecloud.gdapi.model.Schema;
+import io.github.ibuildthecloud.gdapi.model.impl.CollectionImpl;
+import io.github.ibuildthecloud.gdapi.model.impl.WrappedResource;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.request.handler.AbstractApiRequestHandler;
+
+import java.io.IOException;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class DeTransformationHandler extends AbstractApiRequestHandler {
+
+    @Inject
+    TransformerHandler encryptionHandler;
+
+    @Override
+    public void handle(ApiRequest request) throws IOException {
+        if (request.isCommitted() || request.getResponseObject() == null)
+            return;
+        Object response = request.getResponseObject();
+        if (response instanceof WrappedResource) {
+
+            Schema schema = request.getSchemaFactory().getSchema(request.getType());
+            request.setResponseObject(decrypt((Resource) response, schema));
+        } else if (response instanceof CollectionImpl){
+            SchemaFactory schemaFactory = request.getSchemaFactory();
+            for (Resource resource:((CollectionImpl) response).getData()){
+                decrypt(resource, schemaFactory.getSchema(resource.getType()));
+                //ToDo Store to responseObject. Check that the mods apply.
+            }
+            request.setResponseObject(response);
+        }
+    }
+
+    private Resource decrypt(Resource responseObject, Schema schema) {
+        Map<String, Field> fields = schema.getResourceFields();
+        Map<String, Object> object = responseObject.getFields();
+        for (String fieldName: object.keySet()){
+            if (!fields.containsKey(fieldName) || object.get(fieldName) == null){
+                continue;
+            }
+            Field field = fields.get(fieldName);
+            if (StringUtils.isNotBlank(field.getTransform())){
+                String decrypted;
+                try {
+                    decrypted = encryptionHandler.untransform((String) object.get(fieldName));
+                } catch (UnsupportedOperationException | IllegalArgumentException e) {
+                        decrypted = "";
```

Comment:
>> Firstly should always be able to decrypt. -

Would the above be true for the upgrade scenario when field was stored as unencrypted, and then "transform" attribute would be set for it? Just want to make sure that these 2 exceptions resulting in empty return, won't happen in this situation.